### PR TITLE
fix: wrap classifier batch inference in torch.no_grad() to stop memory leak

### LINKED ISSUE #609: Memory usage issues with TIMM classifier
### Search before asking

- [x] I have searched the Pytorch-Wildlife [issues](https://github.com/microsoft/CameraTraps/issues) and found no similar bug report.

### Bug

I found this when trying to use the Deep Faun New England classifier for species classification.  As the code is currently written, running the DFNE classifier (and presumambly the Deep Faun classifier as well) causes memory usage to balloon.  I did a batch classification of approximately 150 images and memory usage (RAM or GPU RAM, doesn't matter) just keeps growing.  I tried running classification on roughly 3000 images and memory usage ballooned to over 300 GB and I didn't even get half way through the set of images.

I believe the issue is in 
PytorchWildlife/models/classification/timm_base/base_classifier.py

in the batch_image_classification function specifically these lines of code:

```
with tqdm(total=len(dataloader)) as pbar: 
            for batch in dataloader:
                imgs, paths = batch
                imgs = imgs.to(self.device)
                total_logits.append(self.predictor(imgs))
                total_paths.append(paths)
                pbar.update(1)
```

I believe the issue is that model is still keeping track of gradient information.  When I modified the code to be as follows:
```
with tqdm(total=len(dataloader)) as pbar: 
       with torch.no_grad():
            for batch in dataloader:
                imgs, paths = batch
                imgs = imgs.to(self.device)
                total_logits.append(self.predictor(imgs))
                total_paths.append(paths)
                pbar.update(1)
```

Memory usage was normal and I was able to run the classifier without issue

### Environment

_No response_

### Minimal Reproducible Example

_No response_

### Additional

_No response_

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

### PR DESCRIPTION
**Pull request by StepFPV**
## What
`batch_image_classification()` in both classifier families ran the forward pass without `torch.no_grad()`:
- `PytorchWildlife/models/classification/timm_base/base_classifier.py` (DFNE / Deepfaune)
- `PytorchWildlife/models/classification/resnet_base/base_classifier.py` (Amazon / Opossum / Serengeti)

Since gradients were tracked, PyTorch retained the autograd graph for every batch appended to `total_logits`, so memory grew unbounded over a run instead of staying flat.

Also moved each batch's logits to CPU right after inference (`.cpu()` per batch) instead of holding them all on GPU until the final `torch.cat(...).cpu()`, so peak VRAM no longer scales with dataset size.

## Why
Fixes #609 — reported ~300GB RAM/VRAM growth and a crash before completing inference on ~3000 images. Root cause and reproduction were already diagnosed in the issue; this PR applies the same fix to both classifier base classes since they share the identical pattern.

## Testing
Confirmed both loops are functionally unchanged (same batch order, same result construction) — only the gradient-tracking and CPU-transfer timing changed. Verified `python -c "import ast; ast.parse(open(f).read())"` on both files.

Scope kept to the reported bug only (`batch_image_classification`); `single_image_classification` isn't part of the leak (no accumulation across a loop) so left untouched to keep this PR focused.

## Disclosure
Diagnosis and implementation were done with AI assistance (Claude Code), reviewed and submitted by me.

## Are you willing to submit a PR?
Yes — this is it.

### CODE REVIEW

**Comment by StepFPV** [comment]
@microsoft-github-policy-service agree
